addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshService.java

```diff
@@ -301,18 +313,35 @@ private void schedulePoller(int initialDelay) {
             }
             httpSender.sendAndReceive(pollMsg);
             if (pollMsg.getResponseHeader().getStatusCode() != 200) {
-                LOGGER.warn(
-                        Constant.messages.getString(
-                                "oast.interactsh.error.poll",
-                                Constant.messages.getString("oast.interactsh.error.badHttpCode")));
+                LOGGER.error(
+                        "Error during interactsh poll, due to bad HTTP code {}. Content: {}",
+                        pollMsg.getResponseHeader().getStatusCode(),
+                        pollMsg.getResponseBody().toString());
                 return new ArrayList<>();
             }
+
             JSONObject response = JSONObject.fromObject(pollMsg.getResponseBody().toString());
             if (!response.containsKey("data") || !response.containsKey("aes_key")) {
-                LOGGER.warn(Constant.messages.getString("oast.interactsh.error.poll.badResponse"));
+                LOGGER.warn(
+                        "Bad polling response received from interactsh server. Missing data or aes_key element. HTTP code {}. Content: {}",
+                        pollMsg.getResponseHeader().getStatusCode(),
+                        pollMsg.getResponseBody().toString());
+                return new ArrayList<>();
             }
             String aesKey = response.getString("aes_key");
             JSONArray interactions = response.getJSONArray("data");
+            if (interactions.size() > 0) {
+                LOGGER.info(
```

Comment:
Could pass the level as argument to avoid the duplication. But does it really need to be info?